core/dbt/adapters/base/impl.py

```diff
@@ -1298,6 +1301,57 @@ def render_column_constraint(cls, constraint: ColumnLevelConstraint) -> str:
         else:
             return ""
 
+    @available
+    def render_column_constraint_ddl(self, columns: Dict[str, Dict]) -> List:
+        rendered_column_constraints = []
+        all_constraints = []
+
+        unsupported_contraints = self.list_unsupported_constraints()
+
+        for k, v in columns.items():
+            column_ddl = [f"{v['name']} {v['data_type']}"]
+            for con in v["constraints"]:
+                all_constraints.append(con)
+                if con["type"] not in unsupported_contraints:
+                    constraint = self._parse_column_constraint(con)
+                    column_ddl.append(self.render_column_constraint(constraint))
+            rendered_column_constraints.append(" ".join(column_ddl))
+
+        self.process_constraint_warnings(constraints=all_constraints)
+
+        return rendered_column_constraints
+
+    def list_unsupported_constraints(self) -> List:
+        not_supported = []
+        for k, v in self.constraint_support().items():
+            if v == ConstraintSupport.NOT_SUPPORTED:
+                not_supported.append(k)
+        return not_supported
+
+    def process_constraint_warnings(self, constraints: List[Dict]):
+        constraint_map = self.constraint_support()
+        not_supported = [
+            c["type"]
+            for c in constraints
+            if constraint_map[c["type"]] == ConstraintSupport.NOT_SUPPORTED
+            and c["warn_unsupported"]
+        ]
+        not_enforced = [
+            c["type"]
+            for c in constraints
+            if constraint_map[c["type"]] == ConstraintSupport.NOT_ENFORCED and c["warn_unenforced"]
+        ]
+
+        if not_supported:
+            warn_or_error(ConstraintNotSupported(constraints=not_supported))
+        if not_enforced:
+            warn_or_error(ConstraintNotEnforced(constraints=not_enforced))
+
+    @staticmethod
+    def constraint_support() -> Dict:
+        # TODO: should this instead default all to NOT_ENFORCED
```

Comment:
In the issue the [recommendation](https://github.com/dbt-labs/dbt-core/issues/7067#issuecomment-1447281379) was to default to: 
```
{
"not_null": ConstraintSupport.ENFORCED,
 "primary_key": ConstraintSupport.NOT_ENFORCED,
 "foreign_key": ConstraintSupport.NOT_ENFORCED,
 "unique": ConstraintSupport.NOT_ENFORCED,
 "check": ConstraintSupport.NOT_SUPPORTED,
}
```